types/block.go

```diff
@@ -225,6 +220,7 @@ func (b *Block) fillHeader() {
 	}
 }
 
+// TODO: Move out from 'types' package
```

Comment:
I've also missed this one. Is the plan to fill the DAHeader when calling PutBlock only?

Reply:
Not necessarily. Just in general, that func does things that probably should not be a part of types package

Reply:
I agree. I guess the question is how we want to handle the DAHeader then. One option is to return the DAHeader with PutBlock and not compute it before that.

Reply:
I like that option